pallets/gear-messenger/src/migrations.rs

```diff
@@ -85,30 +86,43 @@ impl<T: Config> OnRuntimeUpgrade for MigrateToV3<T> {
 
     #[cfg(feature = "try-runtime")]
     fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
-        let mut count = v2::Waitlist::<T>::iter().count();
-        count += v2::Dispatches::<T>::iter().count();
-        count += v2::DispatchStash::<T>::iter().inspect(
-            |store| {
-                if store.1.0.context.is_some() {
-                    panic!("Previous context on StoredDispatch in DispatchStash should always be None, but was Some for message id {:?}", store.1.0.message.id());
-                }
-            },
-        ).count();
-
-        Ok((count as u64).encode())
+        let onchain_version = Pallet::<T>::on_chain_storage_version();
+        if onchain_version < 3 {
+            let mut count = v2::Waitlist::<T>::iter().count();
+            count += v2::Dispatches::<T>::iter().count();
+            count += v2::DispatchStashV2::<T>::iter().inspect(
+                |store| {
+                    if store.1.0.context.is_some() {
+                        panic!("Previous context on StoredDispatch in DispatchStash should always be None, but was Some for message id {:?}", store.1.0.message.id());
+                    }
+                },
+            ).count();
+            Ok((count as u64).encode())
```

Comment:
dont you want to return here encoded Option<u64>?

Reply:
Simple way